### PR TITLE
ci: move to Python 3.11 final

### PR DESCRIPTION
**Pull request by browniebroke**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ jobs:
...
       matrix:
         os: [ubuntu, macos]
-        python-version: ['3.7', '3.8', '3.9', '3.10']
-        # test 3.11-dev and pypy on ubuntu only to speed up CI, no reason why macos X pypy should fail separately
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        # test 3.12-dev and pypy on ubuntu only to speed up CI, no reason why macos X pypy should fail separately
         include:
           - os: 'ubuntu'
-            python-version: '3.11-dev'
+            python-version: '3.12-dev'
```
Comment (browniebroke) [inline .github/workflows/ci.yml:23]: Is that enough to add the trove classifier `Programming Language :: Python :: 3.12` already?

**Comment by browniebroke** [comment]
Not sure why the pypy tests are failing, is that a new issue? Latest build from a few months ago seem to pass and nothing changed in this repo. I'm not following too much the pypy project, I guess something changed over there.

**Comment by samuelcolvin** [comment]
I think we should remove 3.12 for now.

Not sure about pypy, I'll try to look soon.

**Comment by browniebroke** [comment]
Ok, I'll remove the last commit where I added 3.12

**Comment by browniebroke** [comment]
Looks like something is messed up with the cache for 3.11 on macOS. If I bypass it, [it works](https://github.com/samuelcolvin/dirty-equals/actions/runs/3897132363/jobs/6654480073#step:6:38), but with it, [`coverage` isn't installed](https://github.com/samuelcolvin/dirty-equals/actions/runs/3897149712/jobs/6654515982#step:6:15)... 🤔 Am I missing something?

**Comment by samuelcolvin** [comment]
Prefix, the cache key with `v2-`, otherwise I'll fix soon.

**Comment by samuelcolvin** [comment]
thanks so much, I'll fix the remaining problems in a new pr.

**Pull request merged**